### ISSUE TITLE
Unable to Retrieve History Data with Historicaldata Tutorial Example

### ISSUE THREAD

**Issue by iago-byte** [open62541/open62541#6273]
## Description
Greetings team,

I'm interested in using historical data functionality in my OPC UA server, and I've attempted to compile and run the "tutorial_server_historicaldata.c" example using version 1.4-rc2. However, upon reviewing the UAExpert log, I encountered the following error:
![HistoricalDataError](https://github.com/open62541/open62541/assets/151661607/7aad7549-a04c-47f0-b92d-fae17c4cf631)

Additionally, I've checked the server log, but there is no information regarding this issue. I've also tried performing the same procedure with versions 1.3.9, 1.3.8, and 1.3.7, and in these versions, the example code seems to work without any issues. 

![HistoricalData](https://github.com/open62541/open62541/assets/151661607/c4f7cc41-da0c-4903-b5fa-cfcb43705ee8)

Could it be that I'm making a mistake in the configuration or in the usage of version 1.4-rc2?

I appreciate any guidance or advice you can provide to help resolve this issue.

Thank you!
 

## Background Information / Reproduction Steps

Used CMake options:

```bash
cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUA_NAMESPACE_ZERO=FULL -DUA_ENABLE_HISTORIZING=ON -DUA_BUILD_EXAMPLES=ON ..
```

## Checklist

Please provide the following information:

 - [x] open62541 Version: v1.3.9 and v1.4.0-rc2
 - [x] Other OPC UA SDKs used (client or server): UAexpert
 - [x] Operating system: Linux
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

**Comment by lwmake** [comment]
I also have this problem with open62541 v1.4.6, but v1.3.14 works normally.

**Comment by MicroKoder** [comment]
in the file ua_client_highlevel.c

change line:
-    item.dataEncoding = UA_QUALIFIEDNAME(0, "");
+    item.dataEncoding = UA_QUALIFIEDNAME(0, NULL);

**Comment by ffreality** [comment]
> in the file ua_client_highlevel.c
> 
> change line: item.dataEncoding = UA_QUALIFIEDNAME(0, ""); to: item.dataEncoding = UA_QUALIFIEDNAME(0, NULL);

yes, figured it out, solved and actually sent a commit about it.
https://github.com/open62541/open62541/commit/5b8e113515cd6361f967253bd56dd6b835fb2d3c

At first they accepted and added it but latest version has same problem.

**Comment by Noface07** [comment]
when I commented out 
```
        if (!setting->historizingBackend.timestampsToReturnSupported(
                    server,
                    setting->historizingBackend.context,
                    sessionId,
                    sessionContext,
                    &nodesToRead[i].nodeId,
                    timestampsToReturn)) {
            response->results[i].statusCode = UA_STATUSCODE_BADTIMESTAMPNOTSUPPORTED;
            continue;
        }
```

in `ua_history_database_default.c` I am getting timestamp as;

![Image](https://github.com/user-attachments/assets/19a50c64-0f3f-44d6-9adf-0ef69a1bcecb)

but when I monitor it in Data Access View I am getting correct source and server timestamps:

![Image](https://github.com/user-attachments/assets/d737eb66-e796-4996-b88e-2c8d0dd44b4d)

**Comment by Noface07** [comment]
for a quick fix I have edited the `ua_services_attribute.c` :

added these 2 lines at 1410 and 1411:

adjustedValue.hasSourceTimestamp = true;
adjustedValue.sourceTimestamp = UA_DateTime_now();

![Image](https://github.com/user-attachments/assets/8ac76ca0-14f2-4775-8c6c-9122e0783221)

**Comment by segotdko** [comment]
> for a quick fix I have edited the `ua_services_attribute.c` :
> 
> added these 2 lines at 1410 and 1411:
> 
> adjustedValue.hasSourceTimestamp = true; adjustedValue.sourceTimestamp = UA_DateTime_now();

It should be enough to make your variable dynamic UA_Server_setVariableNodeDynamic(server, your node, true). [This is valid for v1.4.11](https://github.com/open62541/open62541/blob/v1.4.11/src/server/ua_services_attribute.c#L1450).

**Comment by lwmake** [comment]
Thanks



***@***.***
 
From: segotdko
Date: 2025-07-03 13:58
To: open62541/open62541
CC: lwmake; Comment
Subject: Re: [open62541/open62541] Unable to Retrieve History Data with Historicaldata Tutorial Example (Issue #6273)
segotdko left a comment (open62541/open62541#6273)
for a quick fix I have edited the ua_services_attribute.c :
added these 2 lines at 1410 and 1411:
adjustedValue.hasSourceTimestamp = true; adjustedValue.sourceTimestamp = UA_DateTime_now();
It should be enough to make your variable dynamic UA_Server_setVariableNodeDynamic(server, your node, true). This is valid for v1.4.11.
